rmc/utils/constraint.py

```diff
@@ -1310,45 +1318,350 @@ def create_context_with_oe(
+    """
+    # Get transcripts with evidence of RMC
+    rmc_transcripts = hl.literal(ht.aggregate(hl.agg.collect_as_set(ht.transcript)))
+
+    # Get all QC pass transcripts and outlier transcripts
+    qc_pass_transcripts = get_constraint_transcripts(outlier=False)
+    outlier_transcripts = get_constraint_transcripts(outlier=True)
+
+    # Get all transcripts displayed on browser
+    transcript_ht = gene_model.ht()
+    all_transcripts = transcript_ht.aggregate(
+        hl.agg.collect_as_set(transcript_ht.transcript)
+    )
+    transcripts_no_rmc = qc_pass_transcripts.difference(rmc_transcripts)
+    transcripts_not_searched = all_transcripts.difference(qc_pass_transcripts)
+    ht = ht.select_globals()
+    return ht.annotate_globals(
+        transcripts_not_searched=transcripts_not_searched,
+        transcripts_no_rmc=transcripts_no_rmc,
+        outlier_transcripts=outlier_transcripts,
+    )
+
+
+def reformat_annotations_for_release(freeze: int, overwrite_temp: bool) -> None:
     """
     Reformat annotations in input HT for release.
 
     This reformatting is necessary to load data into the gnomAD browser.
 
     Desired schema:
-    ---------------------------------------
+    ----------------------------------------
+    Global fields:
+        'transcripts_not_searched': set<str>
+        'transcripts_no_rmc': set<str>
+        'outlier_transcripts': set<str>
+    ----------------------------------------
     Row fields:
-        'transcript_id': str
-        'regions': array<struct {
-            'start': int32
-            'stop': int32
-            'observed_missense': int32
-            'expected_missense': float64
-            'chisq': float64
-        }>
-
+        'interval': interval<locus<GRCh37>>
+        'transcript': str
+        'regions': struct {
+            start_coordinate: locus<GRCh37>,
+            stop_coordinate: locus<GRCh37>,
+            start_aa: str,
+            stop_aa: str,
+            obs: int64,
+            exp: float64,
+            oe: float64,
+            chisq: float64
+        }
     ----------------------------------------
-    Key: ['transcript_id']
+    Key: ['interval', 'transcript']
     ----------------------------------------
 
-    :param ht: Input Table.
-    :return: Table with schema described above.
+    :param freeze: RMC data freeze number.
+    :param overwrite_temp: Whether to overwrite temporary data.
+        If False, will read existing temp data rather than overwriting.
+        If True, will overwrite temp data.
+    :return: None; writes Table with desired schema to resource path.
     """
+    ht = rmc_results.versions[freeze].ht()
+
+    # Annotate start and stop coordinates per region
+    ht = ht.annotate(
+        start_coordinate=ht.interval.start,
+        stop_coordinate=ht.interval.end,
+    )
+
+    # Annotate amino acids
+    ht = annot_rmc_with_aa(ht, overwrite_temp)
+
+    # Add region struct
     ht = ht.annotate(
         regions=hl.struct(
```

Comment:
```suggestion
        region=hl.struct(
```

Reply:
Ah it is an array when you collect in the aggregation below this (which, you're right, should be fixed in the schema in the function documentation) but here this is a row-wise annotation summarizing the row which corresponds to a single region right?

Reply:
oh good point, will update!